bin/pyspark

```diff
@@ -49,7 +49,10 @@ if [[ -n "$IPYTHON_OPTS" || "$IPYTHON" == "1" ]]; then
   # If IPython options are specified, assume user wants to run IPython
   # (for backwards-compatibility)
   PYSPARK_DRIVER_PYTHON_OPTS="$PYSPARK_DRIVER_PYTHON_OPTS $IPYTHON_OPTS"
-  PYSPARK_DRIVER_PYTHON="ipython"
+  if [ -x "$(command -v jupyter)" ]; then
+    PYSPARK_DRIVER_PYTHON="jupyter"
+  else
+    PYSPARK_DRIVER_PYTHON="ipython"
```

Comment:
@rekhajoshm I keep thinking the test failures are spurious, and clearly some of them are, but I'm now double-checking that this seemingly simple change is actually right. Do you not need a closing `fi` after this line? not sure if it's implicit but was speculating that maybe this is causing a failure in a mysterious way at the end when Pyspark tests start. Could be a wrong guess.